### ISSUE TITLE
rest api

### ISSUE THREAD

**Issue by vegarringdal** [Wakanda/wakanda-issues#97]
Hi

You dont have any plans to remove the rest api wakanda already have?
One thing wakanda have that is really great is the entityset combined with selection.
So useful when you need to work with large collections, and need to create new from selection. or just to have working selection over very large collection without pager (I hate pagers)

Would like to play around with making a rest api so I can move a big app I created for fun over to Aurelia with my new grid Ive created for that.

But I really dont want to invest any time into it if its going to be a waists of time because entityset etc will be removed

Also wondering if you ever plan to add function to load a collection from a entityset id, if server dont have that entityset it just throw error, would also be great if it was possible to create a entityset id on server side, so you can use rpc/websockets to send that information.

Thx

**Comment by vegarringdal** [comment]
Would really like a response they you guys have no plans to kill some of the features of the rest api.
Working on a rest api atm where I utilize the entityset with the $keepSel & $fromSel, any changes into that will kinda kill what Im trying to do.
Would be so sad if I manage to make this how I want if after a few weeks and cant use it because you remove the current rest api. 😢 

Not all of the rest api is documented in docs, but the old waf helps figuring that part out :-)

Hope for a positive response, I think the entiyset part is what makes wakanda really cool, and would be hard to replace with other db where you really need to work with a lot of entitysets at the same time, and selection is important.

**Comment by hamzahik** [comment]
I can confirm the current REST API will stay in the product (it is even updated with bug fixes and new features from time to time) **but** a simpler restful API will be added on future versions, we don't plan though to make this new API equivalent to the current.

For undocumented REST API features I can't guarantee they will be kept. If you have a specific feature in mind please state it so I can give you a confirmation.

**Comment by vegarringdal** [comment]
@hamzahik thx for the response.

Good to know that its staying
That you are adding alternative rest api is good too, or just extending the old one with some simplified operations could maybe be better too, then you have just a more advanced/developed, making it possible to expand existing 

Atm Im at a very early stage in the datasource Im trying to make, so only know about $keepSel & $fromSel features that isnt very well documented, but looking at the core rest.js file in waf Ill guess Ill figure out the most, if you guys have a internal pdf docs you could mail me than that would be great too.


**-$keepSel**  used with entityset in a orderby request  allows you to send selection with the rest call and you get back the correct rows, and that is so useful when dealing with endeless scrolling paging since I cant use ID as a ref when using large ranges that might not have been loaded on client side.

**-$fromSel** used with entityset and sending a selection with it will create a new entityset, so usefull when you are about to call a entity collection method, and for showing a subset without loosing the old collection

So basically everything that deals with entityset and selection in the rest api :-)

And the $catalog/$info/directory ofc, but most of that is documented.

Also would be great if it was possible to load a collection from a entityset id on server side, and create one, could be very simplified ```loadCollection(entitysetid, table)```, if its not there it return null, and a ```createEntityset(collection, timeout)```, a serverside ```releaseEntityset(entityset)``` could also been useful too.
Maybe Im the only one that really like the entityset ? 
Just important to release when you dont need them, and set better timeout as needed depending on application

**Comment by vegarringdal** [comment]
dunno if its any interest, but datasource api is starting to work 
Its using a striped out aurleia-v-grid, that uses a connection class for calling the datasource [(link to the unmodified grid)](http://aurelia-ui-toolkits.github.io/demo-v-grid/#/about)
A LOT missing still, but its a start 😂 

Video:
https://drive.google.com/file/d/0B_wM5SJf8IXoTjBnM3pIS0RwWnc/view

also showing the network, so you can see it stays withing the page size all the time, an queue the requests so it does not do unnecessary request when getting a element thats not in local cache
